pkg/operator/controller/cruft.go

```diff
@@ -186,6 +192,145 @@ func reconcileServiceAccounts(args *callbacks.ReconcileCallbackArgs) error {
 	return nil
 }
 
+func getSpecVersions(realCrd *extv1.CustomResourceDefinition) (*extv1.CustomResourceDefinitionVersion, *extv1.CustomResourceDefinitionVersion) {
+	var oldVersion *extv1.CustomResourceDefinitionVersion
+	var newVersion *extv1.CustomResourceDefinitionVersion
+	for index, version := range realCrd.Spec.Versions {
+		if version.Name == OldVersion {
+			oldVersion = &realCrd.Spec.Versions[index]
+		}
+		if version.Name == NewVersion {
+			newVersion = &realCrd.Spec.Versions[index]
+		}
+	}
+	return oldVersion, newVersion
+}
+
+func ensureNewVerIsStorage(args *callbacks.ReconcileCallbackArgs, realCrd, newCrd *extv1.CustomResourceDefinition) (*extv1.CustomResourceDefinition, error) {
+	realCrdCopy := realCrd.DeepCopy()
+	oldVersion, newVersion := getSpecVersions(realCrd)
+	if oldVersion != nil {
+		oldVersion.Storage = false
+	}
+	if newVersion == nil {
+		_, newVersion = getSpecVersions(newCrd)
+		realCrd.Spec.Versions = append(realCrd.Spec.Versions, *newVersion)
+	}
+	newVersion.Storage = true
+	if !reflect.DeepEqual(realCrdCopy, realCrd) {
+		err := args.Client.Update(context.TODO(), realCrd)
```

Comment:
The callback code shouldn't have to update the resource directly since the reconcile loop is going to do that.  Can either update the "desiredObject" on "post read" callback or the "currentObject" on pre update.